inc/customizer.php

```diff
@@ -22,42 +29,24 @@ public function __construct() {
 
 		}
 
-		add_action( 'after_setup_theme', array( $this, 'logo' ) );
-		add_action( 'after_setup_theme', array( $this, 'header' ) );
-		add_action( 'after_setup_theme', array( $this, 'background' ) );
-
-		add_action( 'customize_register', array( $this, 'selective_refresh' ), 11 );
-
-		add_action( 'customize_preview_init',  array( $this, 'customize_preview_js' ) );
-
-	}
-
-	/**
-	 * Add custom logo support.
-	 *
-	 * @action after_setup_theme
-	 * @since  1.0.0
-	 */
-	public function logo() {
-
 		/**
-		 * Filter the custom logo args.
-		 *
-		 * @since 1.0.0
+		 * These methods require custom colors to be available.
 		 *
-		 * @var array
+		 * @global Primer_Customizer_Colors primer_customizer_colors
+		 * @since  1.0.0
 		 */
-		$args = (array) apply_filters( 'primer_custom_logo_args',
-			array(
-				'height'      => 100,
-				'width'       => 400,
-				'flex-height' => true,
-				'flex-width'  => true,
-				'header-text' => array( 'site-title', 'site-description' ),
-			)
-		);
+		if ( isset( $GLOBALS['primer_customizer_colors'] ) ) {
 
-		add_theme_support( 'custom-logo', $args );
+			$this->colors = $GLOBALS['primer_customizer_colors'];
```

Comment:
@fjarrett If the following filters need the customizer colors class then they should be moved to that class I think. Removing circular dependencies and globals if we can.